### PR TITLE
feat: add search api

### LINKED ISSUE #24: Implementation of v2 Search
### What

- As a new version of the Search API is developed, this Library should also implement the usage of it.

See: https://wiki.openfoodfacts.org/Open_Food_Facts_Search_API_Version_2

> The existing Open Food Facts search API is outdated and hacky (it was built on top of the OFF web site search form and is unnecessarily convoluted) and does not support some of the requirements for the Project:Personalized_Search (in particular being able to retrieve a given set of products using their barcodes).
> We are thus creating a new Open Food Facts Search API Version 2 that will be simpler but also more powerful.

### Part of
- #39

### PR DESCRIPTION
**Pull request by Benoit382**
### What
Implementation of https://search.openfoodfacts.org/docs
Route : 
- GET /document/{identifier} : SearchApi::getDocument
- GET /search : SearchApi::search
- GET /autocomplete : SearchApi::autocomplete

### Part of 
#24 Work on implementation of Api Search

### CODE REVIEW

**Review by teolemon — APPROVED** [review]
Sorry for the absence of response on your autocomplete question. Did you manage to turn around it ? A taxonomy is "categories" and a possible value is bread.

**Comment by Benoit382** [comment]
Do you want a second review ?

**Comment by teolemon** [comment]
nope that's fine with me 👍 
BTW, do you have the sufficient rights to merge @Benoit382 ?

**Pull request merged**
